yarn-project/archiver/src/archiver/archiver_store.ts

```diff
@@ -347,6 +364,73 @@ export class MemoryArchiverStore implements ArchiverDataStore {
     return Promise.resolve(logs.slice(startIndex, endIndex));
   }
 
+  /**
+   * Gets unencrypted logs based on the provided filter.
+   * @param filter - The filter to apply to the logs.
+   * @returns The requested logs.
+   * @throws If txHash and block range are both defined.
+   */
+  getUnencryptedLogs(filter: LogFilter): Promise<ExtendedUnencryptedL2Log[]> {
```

Comment:
There is quite a bit of conditional logic here leading to many possible scenarios. Can you create some tests around this new filtering in `archiver_store.test.ts`?

Reply:
Added the tests.